src/app/model/ChessEngine.ts

```diff
@@ -68,7 +70,78 @@ export class ChessEngine implements ChessModel {
     return side === Side.WHITE ? Side.BLACK : Side.WHITE;
   }
 
+  private returnWirtualProposedBoardAfterMove(chessboard: Chessboard, squareFrom: Square, squareTo: Square): Chessboard {
+    const wirtualPorposedChessboard = _.cloneDeep(chessboard);
+    const { squaresWithPiece: proposedSquaresWithPieces } = wirtualPorposedChessboard;
+
+    const movedPiece = chessboard.onPositionPiece(squareFrom);
+    const positionFrom = `${squareFrom.column}${squareFrom.row}`;
+    const positionTo = `${squareTo.column}${squareTo.row}`;
+    delete proposedSquaresWithPieces[positionFrom];
+    proposedSquaresWithPieces[positionTo] = movedPiece as Piece;
+
+    return wirtualPorposedChessboard;
+  }
+
+  public static kingPosition(chessboard: Chessboard, kingSide: Side): Square | undefined {
+    const { squaresWithPiece: squaresWithPieces } = chessboard;
+    const kingPositionKey = Object.keys(squaresWithPieces).find((key) => {
+      const isKingName = squaresWithPieces[key].name === 'King';
```

Comment:
Tutaj tak samo, wyciagamy informacje z piece i robimy na nich operacje. Ktora nota bene nam sie przydaje w innych tez miejscach. Moze by do klasy Piece dodac:
```
 equals(anotherPiece: Piece | undefined): boolean {
    return anotherPiece !== undefined && anotherPiece.name === this.name && anotherPiece.name === this.side;
  }
  ``` i tego uzyc :) ?

Reply:
Aaaa.... czyli Tobie chodzi o to, aby dodać to do innej klasy?
No ok, ale to może w nowym PR, please ?

Reply:
Moze byc osobno, ale tez nie przesadzajmy w rozdrabnianiem sie :)

Reply:
będzie w next